cmd/heimdall-token-analytics/community_tokens.go

```diff
@@ -290,7 +310,14 @@ func (s *service) StreamCommunityTokens(ctx context.Context, req *server.Request
 		events := make(chan server.StreamEvent[ta.CommunityToken], 100)
 
 		sendData := func() bool {
-			tokens, err := s.tokenAnalytics.GetCommunityTokensByExternalAddresses(ctx, req.Data.ExternalAddresses, req.Token.GetMasterPublicKey(), req.Data.IncludeTopHolders)
+			var userIdentifier string
+			if nostrToken, ok := req.Token.(server.NostrToken); ok {
+				userIdentifier = nostrToken.GetMasterPublicKey()
+			} else if xcomToken, ok := req.Token.(server.XComToken); ok {
```

Comment:
Can we fit into the same inetface? I mean XComToken to implement NostrToken + those extra fields about verified

Reply:
X doesn't have masterpubkey/deviceKey, nostr doesn't have userId, handle and etc, so is it good idea to use 1 interface?